### DISCUSSION TITLE
Content Range on File and ETag

### DISCUSSION THREAD

**Discussion by SaltyAom** [elysiajs/elysia#371] [General]
This discussion is meant to discuss/propose/modify behavior for Elysia to handle File and Blob to send *Content-Range* header.

The behavior of 0.7:
Elysia will not send the **content-range** if the status code is set to 206, 304, 412, 416, or if the headers explicitly provide the **content-range**.

It's recommended to use [ETag plugin](https://github.com/bogeychan/elysia-etag) to handle the correct status code to avoid **content-range** collision from cache.
```typescript
const handleFile = (response: File | Blob, set?: Context['set']) => {
	const size = response.size
	if (
		(size &&
			set &&
			set.status !== 206 &&
			set.status !== 304 &&
			set.status !== 412 &&
			set.status !== 416) ||
		(!set && size)
	) {
		if (set)
			return new Response(response as Blob, {
				status: set.status as number,
				headers: Object.assign(
					{
						'accept-ranges': 'bytes',
						'content-range': `bytes 0-${size - 1}/${size}`
					},
					set.headers
				)
			})

		return new Response(response as Blob, {
			headers: {
				'accept-ranges': 'bytes',
				'content-range': `bytes 0-${size - 1}/${size}`
			}
		})
	}

	return new Response(response as Blob)
}
```

**Comment by mtt-artis** [comment]
Would it be possible to get the request range header to automatically send the correct chunks ? Like `sendFile()` on express ?

Should the response always be a stream ?

PS: love your work and the DX on Elysia
